presto-main/src/main/java/com/facebook/presto/execution/StageId.java

```diff
@@ -79,7 +79,13 @@ public int getId()
     @JsonValue
     public String toString()
     {
-        return queryId + "." + id;
+        StringBuilder builder = new StringBuilder();
+        return withStringBuilder(builder).toString();
+    }
+
+    public StringBuilder withStringBuilder(StringBuilder builder)
```

Comment:
how about renaming this to --> `toString(StringBuilder builder)`

Reply:
The thing is it is NOT returning a string. Maybe toStringBuilder?

Reply:
IMHO there is no point in returning a StringBuilder or applying a builder like method naming. 

I agree with Nikhil that an overloaded toString(StringBuilder) fits better.

Reply:
+1 that this should not return `StringBuilder`.  Just make them append so there's only one option when you want to serialize them:

```suggestion
    public void toString(StringBuilder builder)
    {
        builder.append(queryId.getId()).append(".").append(id);
    }
```

The no-arg `toString` becomes:

```java
    @JsonValue
    public String toString()
    {
        StringBuilder builder = new StringBuilder();
        appendString(builder);
        return builder.toString();
    }

```

If `toString` sounds bad because it's not returning a string, what about `appendString`?

Reply:
+1 for `appendString` and passing `StringBuilder` for it makes sense.

Reply:
Thanks for all your precious comments. Really appreciate it. Feel free to take another look.